judge/models/tests/util.py

```diff
@@ -0,0 +1,330 @@
+        return {
+            'name': required_kwargs['code'],
+            'description': '',
+            'time_limit': 1,
+            'memory_limit': 65536,
+            'points': 1,
+            'group': 'group',
+        }
+
+    def process_related_objects(self, required_kwargs, defaults):
+        if not isinstance(defaults['group'], ProblemGroup):
+            defaults['group'] = create_problem_group(defaults['group'])
+
+
+create_problem = CreateProblem()
+
+
+class CreateContest(CreateModel):
+    model = Contest
+    m2m_fields = {
+        'organizers': (Profile, 'user__username'),
+        'problems': (Problem, 'code'),
+        'view_contest_scoreboard': (Profile, 'user__username'),
+        'rate_exclude': (Profile, 'user__username'),
+        'private_contestants': (Profile, 'user__username'),
+        'organizations': (Organization, 'name'),
+        'tags': (ContestTag, 'name'),
+        'banned_users': (Profile, 'user__username'),
+    }
+    required_fields = ('key',)
+
+    def get_defaults(self, required_kwargs, kwargs):
+        _now = timezone.now()
+        return {
+            'name': required_kwargs['key'],
+            'description': '',
+            'start_time': _now - timezone.timedelta(days=100),
+            'end_time': _now + timezone.timedelta(days=100),
+        }
+
+
+create_contest = CreateContest()
+
+
+class CreateContestParticipation(CreateModel):
+    model = ContestParticipation
+    required_fields = ('contest', 'user')
+
+    def process_related_objects(self, required_kwargs, defaults):
+        if not isinstance(required_kwargs['contest'], Contest):
+            required_kwargs['contest'] = create_contest(required_kwargs['contest'])
+        if not isinstance(required_kwargs['user'], Profile):
+            required_kwargs['user'] = create_user(required_kwargs['user']).profile
+
+
+create_contest_participation = CreateContestParticipation()
+
+
+class CreateContestProblem(CreateModel):
+    model = ContestProblem
+    required_fields = ('contest', 'problem')
+
+    def get_defaults(self, required_kwargs, kwargs):
+        return {
+            'points': 100,
+            'order': 1,
+        }
+
+    def process_related_objects(self, required_kwargs, defaults):
+        if not isinstance(required_kwargs['contest'], Contest):
+            required_kwargs['contest'] = create_contest(required_kwargs['contest'])
+        if not isinstance(required_kwargs['problem'], Problem):
+            required_kwargs['problem'] = create_problem(required_kwargs['problem'])
+
+
+create_contest_problem = CreateContestProblem()
+
+
+class CommonDataMixin:
+    fixtures = ['language_all.json']
+
+    @classmethod
+    def setUpTestData(self):
+        self.users = {
+            'superuser': create_user(
+                username='superuser',
+                is_superuser=True,
+                is_staff=True,
+            ),
+            'staff_blogpost_edit_own': create_user(
```

Comment:
Is this user relevant outside of blog testing, for example? Maybe it should be more limited in scope. Similar issues apply to the other users.

Reply:
I've moved most of the users to their respective test case. The users with problem permissions are used in both `ProblemTestCase` and `SubmissionTestCase`. I'm not sure if these users should be duplicated in both test cases or kept in `CommonDataMixin`.